lib/network/http/utils.ts

```diff
@@ -72,8 +79,8 @@ function getByteRangeFromSpec(rangeSpec, objectSize) {
     if (rangeSpec.start < objectSize) {
         // test is false if end is undefined
         return { range: [rangeSpec.start,
-            (rangeSpec.end < objectSize ?
-                rangeSpec.end : objectSize - 1)] };
+            (rangeSpec.end && rangeSpec.end < objectSize ?
+            rangeSpec.end : objectSize - 1)] };
```

Comment:
Just wondering if we can make this more clear with parenthesis?

```suggestion
            ((rangeSpec.end && rangeSpec.end < objectSize) ?
            rangeSpec.end : objectSize - 1)] };
```

Reply:
Sure